diff --git a/dist/index.js b/dist/index.js
index 2eb23eb2503d534e36c40f0578ced4d904960418..7be6c128a72217779dea76aba09f04bec6e90d2e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -21,7 +21,7 @@ const tanstackViteConfig = (options) => {
 			}),
 			options.tsconfigPath ? tsconfigPaths({ projects: [options.tsconfigPath] }) : void 0,
 			dts({
-				outDir: `${outDir}/esm`,
+				outDirs: `${outDir}/esm`,
 				entryRoot: options.srcDir,
 				include: options.srcDir,
 				exclude: options.exclude,
@@ -48,7 +48,7 @@ const tanstackViteConfig = (options) => {
 				}
 			}),
 			cjs ? dts({
-				outDir: `${outDir}/cjs`,
+				outDirs: `${outDir}/cjs`,
 				entryRoot: options.srcDir,
 				include: options.srcDir,
 				exclude: options.exclude,
@@ -87,7 +87,8 @@ const tanstackViteConfig = (options) => {
 					return "esm/[name].js";
 				}
 			},
-			rolldownOptions: { output: { preserveModules: true } }
+			rolldownOptions: { output: { preserveModules: true } },
+			rollupOptions: { output: { preserveModules: true } }
 		}
 	});
 };
